lib/internal/http2/compat.js

```diff
@@ -606,19 +601,19 @@ class Http2ServerResponse extends Stream {
 
     const stream = this[kStream];
     if (!stream.headersSent)
-      this.writeHead(this[kState].statusCode);
+      this.writeHead(state.statusCode);
     return stream.write(chunk, encoding, cb);
   }
 
   end(chunk, encoding, cb) {
-    const stream = this[kStream];
     const state = this[kState];
 
-    if ((state.closed || state.ending) &&
-        state.headRequest === stream.headRequest) {
-      return false;
+    if (state.ending) {
```

Comment:
Per http1 implementation the only time we early out is whether this method has been called previously or not.